src/evotorch/logging.py

```diff
@@ -88,6 +97,315 @@ def _log(self, status: dict):
         raise NotImplementedError
 
 
+class PicklingLogger(Logger):
+    """
+    A logger which periodically pickles the current result of the search.
+
+    The pickled data includes the current center solution and the best solution
+    (if available). If the problem being solved is a reinforcement learning
+    task, then the pickled data also includes the observation normalization
+    data and the policy.
+    """
+
+    def __init__(
+        self,
+        searcher: SearchAlgorithm,
+        *,
+        interval: int,
+        directory: Optional[Union[str, pathlib.Path]] = None,
+        prefix: Optional[str] = None,
+        zfill: int = 6,
+        items_to_save: Union[str, Iterable] = (
+            "center",
+            "best",
+            "pop_best",
+            "median_eval",
+            "mean_eval",
+            "pop_best_eval",
+        ),
+        make_policy_from: Optional[str] = None,
+        after_first_step: bool = False,
+        verbose: bool = True,
+    ):
+        """
+        `__init__(...)`: Initialize the PicklingLogger.
+
+        Args:
+            searcher: The evolutionary algorithm instance whose progress
+                is to be pickled.
+            interval: Expected as an integer n.
+                Pickling is to be done at every n iterations.
+            directory: The directory into which the pickle files will be
+                placed. If given as None, the current directory will be
+                used.
+            prefix: The prefix to be used when naming the pickle file.
+                If left as None, then the prefix will contain the date,
+                time, name of the class of the problem object, and the PID
+                of the current process.
+            zfill: When naming the pickle file, the generation number will
```

Comment:
This feature is useful, but the argument name is a bit odd to me. Is it inspired by a similar name somewhere else @engintoklu ? Maybe we can think of an alternative @flukeskywalker @Higgcz

Reply:
Hi @NaturalGradient, 
The name comes from this method of `str`: https://docs.python.org/3/library/stdtypes.html?highlight=zfill#str.zfill . It is this `zfill` method that is internally used by `PicklingLogger` for formatting the generation number. Therefore, this keyword argument represents this question: _how should `zfill` be applied on the generation number_.